src/praisonai-sandbox/tests/test_docker_filesystem.py

```
@@ -154,3 +154,35 @@ async def check(sandbox):
+
+
+@needs_docker
+def test_a_timed_out_execution_does_not_leave_a_container():
+    """`proc.kill()` ends the docker CLI while the container it started keeps
+    running. It had no --name either, so it got a random Docker name that
+    neither the timeout handler nor `praisonai managed ps` could find."""
+    import asyncio
+
+    from praisonaiagents.sandbox import SandboxConfig
+
+    before = subprocess.run(["docker", "ps", "-q"], capture_output=True,
```

Ruff (0.16.1)

[error] Starting a process with a partial executable path

(S607)